src/Faithlife.WebRequests/Json/AutoWebServiceResponse.cs

```diff
@@ -10,6 +10,21 @@ namespace Faithlife.WebRequests.Json
 	/// </summary>
 	public abstract class AutoWebServiceResponse
 	{
+		/// <summary>
+		/// Gets the HTTP headers on the response.
+		/// </summary>
+		public HttpResponseHeaders Headers { get; private set; }
+
+		/// <summary>
+		/// Gets whether or not the response's content was able to be mapped to a strongly typed property corresponding to the HTTP status code.
+		/// </summary>
+		public bool IsStatusCodeHandled { get; private set; }
+
+		/// <summary>
+		/// Gets the HTTP status code on the response.
+		/// </summary>
+		public HttpStatusCode StatusCode => m_responseStatusCode ?? throw new InvalidOperationException("Response data has not yet been populated.");
```

Comment:
Similarly, I'd be fine allowing these properties on the derived type. We could suppress the unhandled status code exception if one or both of them is specified, which has the additional advantage of not introducing a breaking change.